activeweb/src/main/java/org/javalite/activeweb/SessionFacade.java

```diff
@@ -64,44 +72,32 @@ public Object get(String name){
         return get(name) != null? (T)get(name) : null;
     }
 
-
     /**
      * Removes object from session.
      * @param name name of object
      */
     public Object remove(String name){
         return remove((Object)name);
-
-//        Object val = get(name);
-//        RequestContext.getHttpRequest().getSession(true).removeAttribute(name);
-//        return val;
-    }
-
-    /**
-     * Add object to a session.
-     * @param name name of object
-     * @param value object reference.
-     */
-    public Object put(String name, Serializable value){
-        Object val = RequestContext.getHttpRequest().getSession(true).getAttribute(name);
-        RequestContext.getHttpRequest().getSession(true).setAttribute(name, value);
-        return val;
     }
 
     /**
      * Returns time when session was created. 
      *
-     * @return time when session was created.
+     * @return time when session was created or -1 if session is not exists.
      */
     public long getCreationTime(){
-        return RequestContext.getHttpRequest().getSession(true).getCreationTime();
+        HttpSession session = httpSession();
+        return session == null ? -1 : session.getCreationTime();
     }
 
     /**
      * Invalidates current session. All attributes are discarded.
      */
     public void invalidate(){
-        RequestContext.getHttpRequest().getSession(true).invalidate();
+        HttpSession session = httpSession();
```

Comment:
Use `exists()` in all cases like this

Reply:
We can lost the session between two requests to RequestContext. I think this version is more safe.